options.go

```diff
@@ -93,6 +93,14 @@ func WithDisableQuerySpanNamePrefix() Option {
 	})
 }
 
+// WithDisableConnectionDetailsInAttributes will disable logging the connection details.
+// in the query span's attributes. It will still be logged in the connect span.
+func WithDisableConnectionDetailsInAttributes() Option {
```

Comment:
I have some issues with this. First it's not clear from the option naming that it's disabled for everything except for a `TraceConnectStart`. Also I'd rather not make assumptions when the connection details are useful for a user of this library or when it's not. While it's noise for one user, it might be very important to another.

If we really want this option, we should check it on every operation (where applicable) so there is at least less ambiguity.

Reply:
So you mean remove the exception simply not log it anywhere?

Reply:
I mean we also add it to `TraceConnectStart` so it's consistent across all operations: we don't log connection details at all, if this option is passed

Reply:
Or perhaps it makes sense to have multiple booleans for the different trace types?

Reply:
Honestly I don't see the benefit of complicating the API surface for what I would see a convenience change. I would either apply it to all operations, or don't apply it all.